### PR TITLE
feat: add goal-question-metric example

### PR DESCRIPTION
**Pull request by jeffabailey**

### CODE REVIEW

File: measuring/gqm_example/gqm_example_metric.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Example Metric
+
+**A short name of the metric**
+
+A brief description of the metric.
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_metric.md:5]: @gruetter suggests: How to interpret that metric is very important!

Comment (gruetter) [inline measuring/gqm_example/gqm_example_metric.md:5]:
```suggestion
A brief description of the metric.

- **Synopsis**: Brief summary of the metric
- **Scale/UoM**: What is the scale and unit of measure of the metric? (nominal, ordinal, interval, ratio - the scale limits the types of operations you can do. E. g. you can't compute averages on a nominal scale, even if the ordinals are expressed as numbers)
- **Interpretation**: What do different values or value ranges mean? (good/bad, little/a lot, ...)
-**Measuring**: How do you actually calculate the value of the metric and  what are the input sources (I think that is what _pseudo-code_ is referring to, below).
```

File: README.md [hunk 2]
```diff
@@ -26,8 +26,19 @@ This book is open to anyone interested in InnerSource, either you are
...
 
+We document each goal in a separate folder in the [measuring](./measuring) directory. Goals link to relevant questions and metrics in the same directory.
+
+*Examples:*
+
+* [Example Goal](./measuring/gqm_example/gqm_example_goal.md)
+* [Example Question](./measuring/gqm_example/gqm_example_question.md)
+* [Example Metric](./measuring/gqm_example/gqm_example_metric.md)
+
+We interconnect our goals, questions, and metrics. This approach not only expands our documentation of these aspects but also keeps it well-organized.
```
Comment (jeffabailey) [inline README.md:41]: Do we need additional examples to illustrate the view we plan to create?

Here's a view when adding a leaf question that relates to the **gqm_example**.

image

I don't want to overcomplicate the example(s), but see value in demonstrating a few use cases.

Comment (rrrutledge) [reply README.md:41]: I will try to spend time this week adding some concrete examples.

File: measuring/gqm_example/gqm_example_goal.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Example Goal
+
+**A short name of the goal**
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_goal.md:3]:
```suggestion
# Short name of the goal
```

File: measuring/gqm_example/gqm_example_question.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# Question Example
+
+**The question**
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_question.md:3]:
```suggestion
# The question

A brief description of the question.
```

File: measuring/gqm_example/gqm_example_question.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+## Goals
+
+[Example Goal](./gqm_example_goal.md)
+
+## Metrics
+
+[Example Metric](./gqm_example_metric.md)
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_question.md:11]:
```suggestion
## Goals

* [Example Goal](./gqm_example_goal.md)

## Metrics

* [Example Metric](./gqm_example_metric.md)
```

File: measuring/gqm_example/gqm_example_goal.md [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+
+A brief paragraph describing the goal and why it matters.
+
+## Metrics
+
+* [Example Metric](./gqm_example_metric.md)
+
+## Questions
+
+* [Example Question](./gqm_example_question.md)
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_goal.md:13]:
```suggestion
## Questions

* [Example Question](./gqm_example_question.md)

## Metrics

* [Example Metric](./gqm_example_metric.md)
```

Comment (rrrutledge) [reply measuring/gqm_example/gqm_example_goal.md:13]: Keep the G-Q-M ordering.

File: measuring/gqm_example/gqm_example_metric.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_metric.md:5]:
```suggestion
# Short name of the metric
```

File: measuring/gqm_example/gqm_example_metric.md [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+
+**Synopsis**: A brief summary of the metric.  
+
+**Scale/UoM**: What scale and unit of measure does the metric use? The scale determines what kinds of operations are possible. For example, you cannot calculate averages on a nominal scale, even if the ordinals are expressed as numbers. The scale types are nominal, ordinal, interval, and ratio.
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_metric.md:9]: What does this mean?  We need to explain it simpler.

File: measuring/gqm_example/gqm_example_metric.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+**Synopsis**: A brief summary of the metric.  
+
+**Scale/UoM**: What scale and unit of measure does the metric use? The scale determines what kinds of operations are possible. For example, you cannot calculate averages on a nominal scale, even if the ordinals are expressed as numbers. The scale types are nominal, ordinal, interval, and ratio.  
+
+**Interpretation**: What do the different values or ranges signify? For example, what does "good" or "bad" mean, or how much is "little" versus "a lot"?  
+
+**Measuring**: What input sources do you use to calculate the metric's value?  
+
+```Pseudo-code for the metric```
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_metric.md:15]:
```suggestion
**Measuring**

What input sources do you use to calculate the metric's value?  

```Pseudo-code for the metric```
```

File: measuring/gqm_example/gqm_example_metric.md [hunk 10]
```diff
@@ -0,0 +1,23 @@
...
+
+## Goals
+
+[Example Goal](./gqm_example_goal.md)
+
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_metric.md:20]:
```suggestion
```

File: measuring/gqm_example/gqm_example_goal.md [hunk 11]
```diff
@@ -0,0 +1,13 @@
+# Example Goal
+
+**A short name of the goal**
+
+A brief paragraph describing the goal and why it matters.
+
+## Metrics
+
+* [Example Metric](./gqm_example_metric.md)
+
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_goal.md:10]:
```suggestion
```

**Review by rrrutledge — CHANGES_REQUESTED** [review]
Thanks for this work, Jeff!  I tried to use these templates and came up with https://github.com/InnerSourceCommons/managing-inner-source-projects/pull/23 and also some feedback on the templates themselves.  Take a look and see what you think!

File: measuring/gqm_example/gqm_example_metric.md (hunk 8 above)
Comment (jeffabailey) [reply measuring/gqm_example/gqm_example_metric.md:7]: I tried making @gruetter 's guidance easier to understand, but maybe it's more complicated?

Here's the original content.

**Scale/UoM**: What is the scale and unit of measure of the metric? (nominal, ordinal, interval, ratio - the scale limits the types of operations you can do. E. g. you can't compute averages on a nominal scale, even if the ordinals are expressed as numbers)

Maybe we need a diagram?

```mermaid
graph TD;
A[Scale/UoM] --> B[What is the scale and unit of measure of the metric?];
B --> C1[nominal];
B --> C2[ordinal];
B --> C3[interval];
B --> C4[ratio];
C1 --> D1[Can't compute averages even if expressed as numbers];
```

Maybe we need something like this?

image

Comment (rrrutledge) [reply measuring/gqm_example/gqm_example_metric.md:7]: Oh I see it was some text from @gruetter.  I went and looked it up, but not sure what point of confusion are we trying to solve by including this section?  Thanks for helping me to understand.

Comment (jeffabailey) [reply measuring/gqm_example/gqm_example_metric.md:7]: I haven't encountered confusion around this, but I presume @gruetter has.

I wonder if there's a middle-ground where we define the metric data type and let people consider details about the unit of measure themselves?

File: measuring/gqm_example/gqm_example_goal.md [hunk 12]
```diff
@@ -0,0 +1,11 @@
+# **Goal:** Short name of the goal
+
+A brief paragraph describing the goal and why it matters.
+
+## Questions
+
+* [Example Question](./gqm_example_question.md)
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_goal.md:7]: In addition to linking them, explain why they are related.

**Comment by jeffabailey** [comment]
TODO: 

- [ ] Add guidance on when to create relationships.
- [ ] Explain why the relationship exists.

File: measuring/gqm_example/gqm_example_goal.md (hunk 12 above)
Comment (rrrutledge) [reply measuring/gqm_example/gqm_example_goal.md:7]: Let's have the reviewers think through this!

File: measuring/gqm_example/gqm_example_goal.md [hunk 13]
```diff
@@ -0,0 +1,11 @@
...
+ 
+## Metrics
+
```
Comment (rrrutledge) [inline measuring/gqm_example/gqm_example_goal.md:10]:
```suggestion
# **Goal:** Short name of the goal

A brief paragraph describing the goal and why it matters.

## Related Questions

* [Example Question](./gqm_example_question.md)
```

**Comment by rrrutledge** [comment]
We need to document the high-level flows for using metrics in InnerSource.
Who are the stakeholders that use them?

* Executive level uses them to see if InnerSource is "working"
* InnerSource Program Office uses them to see how to improve the program.
* InnerSouce Project Drivers use them to advocate for their project and to run their project.

**Comment by rrrutledge** [comment]
At the root README we want some context on how to use this thing and sample user journeys.

**Pull request merged**